nsqadmin/static/js/views/channel.js

```diff
@@ -0,0 +1,47 @@
+var $ = require('jquery');
+
+window.jQuery = $;
+var bootstrap = require('bootstrap'); //eslint-disable-line no-unused-vars
+var bootbox = require('bootbox');
+
+var Pubsub = require('../lib/pubsub');
+var AppState = require('../app_state');
+
+var BaseView = require('./base');
+
+var ChannelView = BaseView.extend({
+    className: 'channel container-fluid',
+
+    template: require('./spinner.hbs'),
+
+    events: {
+        'click .channel-actions button': 'channelAction'
+    },
+
+    initialize: function() {
+        BaseView.prototype.initialize.apply(this, arguments);
+        this.listenTo(AppState, 'change:graph_interval', this.render);
+        this.model.fetch().done(function() {
```

Comment:
Another way to handle this is to listen to `this.model`'s `sync` event and use `render` as the callback.

You also might be able to call `render` when this view is instantiated - placing a loader in the view if the model has no data yet. Then when the fetch returns, the render callback mentioned above will fire - filling the view with the model's newly fetched data.